vector/src/main/java/im/vector/app/features/notifications/FilteredEventDetector.kt

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package im.vector.app.features.notifications
+
+import im.vector.app.ActiveSessionDataSource
+import im.vector.app.features.voicebroadcast.isVoiceBroadcast
+import im.vector.app.features.voicebroadcast.sequence
+import org.matrix.android.sdk.api.session.events.model.isVoiceMessage
+import org.matrix.android.sdk.api.session.getRoom
+import org.matrix.android.sdk.api.session.room.getTimelineEvent
+import org.matrix.android.sdk.api.session.room.model.message.asMessageAudioEvent
+import org.matrix.android.sdk.api.session.room.timeline.TimelineEvent
+import javax.inject.Inject
+
+class FilteredEventDetector @Inject constructor(
+        private val activeSessionDataSource: ActiveSessionDataSource
+) {
+
+    /**
+     * Returns true if the given event should be ignored.
+     * Used to skip notifications if a non expected message is received.
+     */
+    fun shouldBeIgnored(notifiableEvent: NotifiableEvent): Boolean {
+        val session = activeSessionDataSource.currentValue?.orNull() ?: return false
+
+        if (notifiableEvent is NotifiableMessageEvent) {
+            val room = session.getRoom(notifiableEvent.roomId) ?: return false
+            val timelineEvent = room.getTimelineEvent(notifiableEvent.eventId) ?: return false
```

Comment:
If each of these might be null and fall back to returning false, do we risk dropping notifications unnecessarily?

Would it be safer to fall back to returning true?

Reply:
I think this is fine, returning false means **not** ignore the notification, so it should not drop the notification in this case, wdyt?

Reply:
I did the same as https://github.com/vector-im/element-android/blob/f4b7161db216991a94f1a93a2391d77511fbbf7f/vector/src/main/java/im/vector/app/features/notifications/OutdatedEventDetector.kt#L31, in this detector, messages are not considered as outdated if the room or session is not retrieved

Reply:
That makes sense - I misread the method name. I've edited my comment and approved